src/Stratis.Features.FederatedPeg/TargetChain/SignedMultisigTransactionBroadcaster.cs

```diff
@@ -110,5 +110,80 @@ public void Stop()
                 this.signals.Unsubscribe(this.onCrossChainTransactionFullySignedSubscription);
             }
         }
+
+        private async Task OnCrossChainTransactionFullySignedAsync(CrossChainTransferTransactionFullySigned @event)
+        {
+            if (this.ibdState.IsInitialBlockDownload() || !this.federationWalletManager.IsFederationWalletActive())
+            {
+                this.logger.LogInformation("Federation wallet isn't active or the node is IBD.");
+                return;
+            }
+
+            await BroadcastFullySignedTransfersAsync(@event.Transfer);
+        }
+
+        private async Task<SignedMultisigTransactionBroadcastResultItem> BroadcastFullySignedTransfersAsync(ICrossChainTransfer crossChainTransfer)
+        {
+            var transferItem = new SignedMultisigTransactionBroadcastResultItem()
+            {
+                DepositId = crossChainTransfer.DepositTransactionId.ToString(),
+                TransactionId = crossChainTransfer.PartialTransaction.GetHash().ToString(),
+            };
+
+            TxMempoolInfo txMempoolInfo = await this.mempoolManager.InfoAsync(crossChainTransfer.PartialTransaction.GetHash()).ConfigureAwait(false);
+            if (txMempoolInfo != null)
+            {
+                this.logger.LogInformation("Deposit '{0}' already in the mempool.", crossChainTransfer.DepositTransactionId);
+                transferItem.ItemMessage = $"Deposit '{crossChainTransfer.DepositTransactionId}' already in the mempool.";
+                return transferItem;
+            }
+
+            this.logger.LogInformation("Broadcasting deposit '{0}', a signed multisig transaction '{1'} to the network.", crossChainTransfer.DepositTransactionId, crossChainTransfer.PartialTransaction.GetHash());
+
+            await this.broadcasterManager.BroadcastTransactionAsync(crossChainTransfer.PartialTransaction).ConfigureAwait(false);
+
+            // Check if transaction was added to a mempool.
+            TransactionBroadcastEntry transactionBroadCastEntry = this.broadcasterManager.GetTransaction(crossChainTransfer.PartialTransaction.GetHash());
+            if (transactionBroadCastEntry?.TransactionBroadcastState == TransactionBroadcastState.CantBroadcast && !CrossChainTransferStore.IsMempoolErrorRecoverable(transactionBroadCastEntry.MempoolError))
+            {
+                this.logger.LogWarning("Deposit '{0}' rejected: '{1}'.", crossChainTransfer.DepositTransactionId, transactionBroadCastEntry.ErrorMessage);
+                this.crossChainTransferStore.RejectTransfer(crossChainTransfer);
+                transferItem.ItemMessage = $"Deposit '{crossChainTransfer.DepositTransactionId}' rejected: '{transactionBroadCastEntry.ErrorMessage}'.";
+                return transferItem;
+            }
```

Comment:
Perhaps make this a separate method and also check it before calling this method from `BroadcastFullySignedTransfersAsync`.